src/txn/txn_rollback_to_stable.c

```diff
@@ -508,11 +508,10 @@ __rollback_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_REF *ref, WT_PAGE *page
             WT_STAT_CONN_DATA_INCR(session, txn_rts_hs_restore_updates);
 
             /*
-             * We have a tombstone on the original update chain and it is behind the stable
-             * timestamp, we need to restore that as well.
+             * We have a tombstone on the original update chain and it is behind or equal to the
+             * stable timestamp, we need to restore that as well.
              */
-            if (hs_stop_durable_ts <= rollback_timestamp &&
-              hs_stop_durable_ts < newer_hs_durable_ts) {
+            if (hs_stop_durable_ts <= rollback_timestamp) {
```

Comment:
Can we add some sorts of assert here to check the on disk value is an out of order prepared update and in the other cases hs_stop_durable_ts < newer_hs_durable_ts should be true?

Reply:
Yes, the assert is added to verify the restored tombstone timestamp must be less than the newer update timestamp or an out of order prepared update.